cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -560,6 +560,87 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
                     Opcode::Imul => match ty {
                         types::I16X8 => SseOpcode::Pmullw,
                         types::I32X4 => SseOpcode::Pmulld,
+                        types::I64X2 => {
+                            // We will compute for each lane element:
+                            // Ah_Al * Bh_Bl = (Al * Bl) + (Al * Bh) + (Ah * Bl)
+                            // This algorithm will use pmuldq which operates on
+                            // the lower 32-bit (Al or Bl) of a lane and writes the result
+                            // to the full 64-bits of the lane of the destination. For this
+                            // reason we don't need a mask to isolate the lower 32-bits, but
+                            // we did need to shift to isolate the high 32-bits when doing
+                            // calculations.
+                            //
+                            //The sequence is as follows where A=lhs and B=rhs cl operands:
+                            //
+                            // A' = A
+                            // A'' = A
+                            // A = Al * Bl
+                            // A' = A' -> 32
+                            // A' = Ah' * Bl
+                            // A' = A' <- 32
+                            // A = A + A'
+                            // B = B -> 32
+                            // B = Bh * Al''
+                            // B = B <- 32
+                            // A = A + B
```

Comment:
Can I recommend to also use the actual register names for the summary, here, that is: `dst` when it's used, in place of `A`, etc.?

Reply:
In addition to the above comment (which I still think would be nice to address, please :)), I have another suggestion that would spare one right shift.

The current decomposition does `(Ah * Bl) << 32 + (Al * Bh) << 32 + Al * Bl`. I would suggest to factor out the `<< 32`, so have `(Ah * Bl) + (Al * Bh)` and then apply the left shift to this; it would spare one left shift operation. Does it make sense? (This is how Spidermonkey does it, fwiw.)